### PR TITLE
Make bash autocompletion dynamic (#187).

### PR DESCRIPTION
**Pull request by stweller**
OK, here is a more dynamic version. It can complete task id's and lists. In theory it could also complete locations, however location is not present in "porcellain" output. I think it should definitely be there, should I open an issue for that?

### CODE REVIEW

File: contrib/completion/bash/_todo [hunk 1]
```diff
@@ -1,3 +1,21 @@
+_todo_get_ids()
+{
+	local pattern='^\s*"id": \([0-9]\+\),'
+	todo --porcelain list | grep -o "${pattern}" | sed -e "s/${pattern}/\1/" | sort | uniq
```
Comment (hobarrera) [inline contrib/completion/bash/_todo]: Is `uniq` necesary here?

**Comment by hobarrera** [comment]
Looks like your email in your commits doesn't match an email on your github account, so your profile isn't linked it commit authorship pages, etc.
Dunno if you care about that, but just pointing that out (note: I completely don't mind though 🙂 )

File: contrib/completion/bash/_todo (hunk 1 above)
Comment (stweller) [reply contrib/completion/bash/_todo]: No, you're right, that can be omitted here. `id` should be unique anyway.

**Comment by stweller** [comment]
I updated my github profile btw, so the commits should now be linked correctly.

**Comment by hobarrera** [comment]
Thanks!

**Pull request merged**